templates/events-list.php

```diff
@@ -6,41 +6,75 @@
 
 
 
-<h2 class="event_page_title">Upcoming Translation Events</h2>
+<h2 class="event_page_title">Translation Events</h2>
 <div class="event-left-col">
 <?php
-if ( $query->have_posts() ) :
+if ( $current_events_query->have_posts() ) :
 	?>
+	<h3>Current events</h3>
 	<ul>
 		<?php
-		while ( $query->have_posts() ) :
-			$query->the_post();
-			$event_start = get_post_meta( get_the_ID(), '_event_start', true );
+		while ( $current_events_query->have_posts() ) :
+			$current_events_query->the_post();
+			$event_start = ( new DateTime( get_post_meta( get_the_ID(), '_event_start', true ) ) )->format('l, F j, Y');
 			?>
 			<li class="event-list-item">
-				<span class="event-list-date"><time class="event-utc-time" datetime="<?php echo esc_attr( $event_start ); ?>"></span>
-				<a href="<?php echo esc_url( gp_url( wp_make_link_relative( get_the_permalink() ) ) ) ?>"><?php the_title(); ?></a> by <span><?php the_author(); ?></span>
+				<span class="event-list-date"><?php echo esc_html( $event_start ); ?></span>
+				<a href="<?php the_permalink(); ?>"><?php the_title(); ?></a> by <span><?php the_author(); ?></span>
 				<p><?php the_excerpt(); ?></p>
 			</li>
-			<?php
+		<?php
 		endwhile;
 		?>
 	</ul>
 
 	<?php
-	echo esc_html(
-		paginate_links(
-			array(
-				'total'     => $query->max_num_pages,
-				'current'   => max( 1, get_query_var( 'paged' ) ),
-				'prev_text' => '&laquo; Previous',
-				'next_text' => 'Next &raquo;',
-			)
+	echo paginate_links(
+		array(
+			'total'     => $current_events_query->max_num_pages,
+			'current'   => max( 1, $current_events_query->query_vars['current_events_paged'] ),
+			'format'	=> '?current_events_paged=%#%',
+			'prev_text' => '&laquo; Previous',
+			'next_text' => 'Next &raquo;',
 		)
 	);
 
 	wp_reset_postdata();
-else :
+endif;
+if ( $upcoming_events_query->have_posts() ) :
+	?>
+	<h3>Upcoming events</h3>
+	<ul>
+		<?php
+		while ( $upcoming_events_query->have_posts() ) :
+			$upcoming_events_query->the_post();
+			$event_start = ( new DateTime( get_post_meta( get_the_ID(), '_event_start', true ) ) )->format('l, F j, Y');
+			?>
+			<li class="event-list-item">
+				<span class="event-list-date"><?php echo esc_html( $event_start ); ?></span>
+				<a href="<?php the_permalink(); ?>"><?php the_title(); ?></a> by <span><?php the_author(); ?></span>
```

Comment:
I guess we should use the glotpress link here?

```suggestion
				<a href="<?php echo esc_url( gp_url( wp_make_link_relative( get_the_permalink() ) ) ) ?>"><?php the_title(); ?></a> by <span><?php the_author(); ?></span>?></span>
```

Reply:
I noticed we need the event URL in multiple places, and since it's a long chain of function calls, in the future we could add a function somewhere that takes the event id as argument and returns the URL.

Reply:
Thank you for these updates! Related to these chain of functions, they are in a loop, so we should compute in the `WPORG_GP_Translation_Events_Route` class who calls the current template. But there, in the `events_list` method, we only use 2 `WP_Query` queries to pass them to the template, so we will overengineer the final solution, so for me, it is fine the current approach.